mlrun/runtimes/nuclio/application/application.py

```diff
@@ -349,23 +358,25 @@ def deploy(
         )
 
     def with_source_archive(
-        self, source, workdir=None, pull_at_runtime=True, target_dir=None
+        self,
+        source: str,
+        workdir: str = None,
+        target_dir: str = None,
```

Comment:
These args are not aligned between the same method in different classes.
Doesn't it cause issues?

Reply:
note this is aligned with KubejobRuntime and not with RemoteRuntime (which is not a KubejobRuntime) but it seems to work